test/Jaeger.Core.Tests/Reporters/RemoteReporterTests.cs

```diff
@@ -232,33 +243,51 @@ public void TestFlushUpdatesQueueLength()
 
             Assert.Equal(0, _metricsFactory.GetGauge("jaeger:reporter_queue_length", ""));
 
-            RemoteReporter remoteReporter = (RemoteReporter)_reporter;
-            remoteReporter.Flush();
+            ((RemoteReporter)_reporter).Flush();
 
             Assert.True(_metricsFactory.GetGauge("jaeger:reporter_queue_length", "") > 0);
         }
 
-        [Fact(Skip = "Doesn't yet work properly")]
+        [Fact]
         public void TestFlushIsCalledOnSender()
         {
+            CountdownEvent countdownEvent = new CountdownEvent(1);
```

Comment:
When running alone, this seems to succeed. When I run the whole session, this most of the time fails. Looks like there are some side effects on other tests.

Reply:
I hate multi threading tests! 😞